libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -684,28 +684,36 @@ string YulUtilFunctions::overflowCheckedIntMulFunction(IntegerType const& _type)
 			function <functionName>(x, y) -> product {
 				x := <cleanupFunction>(x)
 				y := <cleanupFunction>(y)
+				product := <cleanupFunction>(mul(x, y))
 				<?signed>
-					// overflow, if x > 0, y > 0 and x > (maxValue / y)
-					if and(and(sgt(x, 0), sgt(y, 0)), gt(x, div(<maxValue>, y))) { <panic>() }
-					// underflow, if x > 0, y < 0 and y < (minValue / x)
-					if and(and(sgt(x, 0), slt(y, 0)), slt(y, sdiv(<minValue>, x))) { <panic>() }
-					// underflow, if x < 0, y > 0 and x < (minValue / y)
-					if and(and(slt(x, 0), sgt(y, 0)), slt(x, sdiv(<minValue>, y))) { <panic>() }
-					// overflow, if x < 0, y < 0 and x < (maxValue / y)
-					if and(and(slt(x, 0), slt(y, 0)), slt(x, sdiv(<maxValue>, y))) { <panic>() }
+					<?256bit>
+						// special case
+						if and(eq(x, sub(0, 1)), eq(y, <minValue>)) { <panic>() }
```

Comment:
Apart from the other comments, I also wonder if
```suggestion
						if and(slt(x, 0), eq(y, <minValue>)) { <panic>() }
```

May be cheaper, resp. result in smaller code (depending on optimizer settings), since it avoids the large constant...

(the change would have this check overlap with the check below, but that doesn't matter)

Reply:
It makes sense. Done.